src/hooks/executor.rs

```diff
@@ -71,6 +71,10 @@ pub async fn executor_run(
         return Ok(());
     }
 
+    // Set DuckDB search path according search path in Postgres
+    // Make sure it could find unqualified relations.
+    set_search_path()?;
```

Comment:
Can you comment on why setting the search path was moved from the FDW to the hook?

Reply:
If my understanding is correct, the code in FDW is to set the duckdb schema for the exector hook. FDW queries are executed with schema qualification. I think we only need to set search path only when pushdown the origin query to duckdb directly.